torch/csrc/distributed/autograd/context/container.cpp

```diff
@@ -237,12 +237,13 @@ void DistAutogradContainer::sendReleaseContextRpc(
           CleanupAutogradContextReq(context_id).toMessage(),
           options);
 
-      cleanupFuture->addCallback([](const rpc::FutureMessage& cleanupFuture) {
+      cleanupFuture->addCallback([worker_id](const rpc::FutureMessage& cleanupFuture) {
         if (cleanupFuture.hasError()) {
           std::string errorMsg = c10::str(
-              "Could not release Dist Autograd Context after ",
-              kNumCleanupContextRetries,
-              " attempts.");
+              "Could not release Dist Autograd Context on node ",
+              worker_id,
+              ": ",
+              error->what());
```

Comment:
where does this error come from?

Reply:
This is when cleaning up the autograd contexts on neighboring nodes fails. It may happen if there's an ungraceful shutdown (as in `test_backward_node_failure`) or if all the ReleaseContextRpc's fail.

Reply:
Changing to `cleanupFuture.error->what()`